### ISSUE TITLE
omi server cannot start: failed to change directory

### ISSUE THREAD

**Issue by kanelo75** [microsoft/omi#729]
I have the following error in the omiserver.log when I start the omi server, and the omi server cannot be started.

failed to change directory to:
EventId=10005 Priority=FATAL abnormal termination of engine process detected...shutting down server

I check the code. It should have the directory name after "failed to change directory to:"

**Comment by JumpingYang001** [comment]
@kanelo75  can you share the whole omiserver.log?
also check these commands: `rpm -qa|grep omi;rpm -qa|grep scx;dpkg -l|grep omi;dpkg -l|grep scx;ps -ef|grep omi;ls -al /opt;ls -al /var/opt/omi/run`

**Comment by JumpingYang001** [comment]
@cinimodmik it seems the program cannot load rundir, can you run `/opt/omi/bin/omiserver -p`?

**Comment by cinimodmik** [comment]
hi @JumpingYang001, i have only running it via scxadmin -start and systemctl start omid. Do i run /opt/omi/bin/omiserver -p directly?

**Comment by JumpingYang001** [comment]
@cinimodmik yes, run `/opt/omi/bin/omiserver -p` directly with root account.

**Comment by cinimodmik** [comment]
@JumpingYang001, will try and get back

**Comment by cinimodmik** [comment]
@JumpingYang001, i have attached the results .
[omilog2.txt](https://github.com/microsoft/omi/files/9680535/omilog2.txt)

**Comment by JumpingYang001** [comment]
@cinimodmik that means root user can get OMI_GetPath, and omi user cannot get OMI_GetPath, it looks strange, I have no idea now.
root user get OMI_GetPath values :https://github.com/microsoft/omi/blob/b133548bc064c5874fdba5f5c6f08d11d78deb39/Unix/server/server.c#L387
https://github.com/microsoft/omi/blob/e4d72481fa2f805148c9c8f4d0183b3e2d7814a8/Unix/base/paths.c#L171 when `/opt/omi/bin/omiserver -p`

BTW, is your box enabled selinux?

**Comment by kanelo75** [comment]
yes, the box is enabled selinux if I am not mistaken.
Is there anyway we can configure those -p information in some configuration file and let's omi user to retrieve the information.

**Comment by JumpingYang001** [comment]
it has built by default, and some values from `/etc/opt/omi/conf/omiserver.conf`.

**Comment by cinimodmik** [comment]
Hi @JumpingYang001,

I have also tried previously to temporally set selinux to permissive but it is still the same failure results.

**Comment by cinimodmik** [comment]
hi @JumpingYang001, any further advise for this issue

**Comment by JumpingYang001** [comment]
@cinimodmik after set selinux t permissive do you reboot the machine?
we didn't repro the issue, so I suggest you to create a ticket on Azure vm or on SCOM support. are you a SCOM customer or Azure customer?

**Comment by cinimodmik** [comment]
Hi @JumpingYang001 , I have not reboot the machine as selinux will be resumed to enforce if i do that.
Noted thanks for your help .

**Comment by kanelo75** [comment]
@JumpingYang001 is the SCOM support you mention under Microsoft? @cinimodmik have raised this case to Microsoft before, and there is one Microsoft engineer has come to look at it, and he seemed to have no idea too. To move it forward, if we raise to Microsoft again, anything we could do in order to speed up to find the root cause.

**Comment by JumpingYang001** [comment]
@kanelo75 yes, SCOM support is under Microsoft.
If that's the case, you should ping the engineer to contact OMI team.
